File: src/main/java/com/lkimilhol/matchingproject/member/domain/Country.java
package com.lkimilhol.matchingproject.member.domain;

import com.lkimilhol.matchingproject.exception.NotFoundCountryException;

public enum Country {
    KR, JA
    ;

    static public Country get(String name) {
        switch (name) {
            case "KR":
                return KR;
            case "JA":
                return JA;
        }

        throw new NotFoundCountryException();
    }
}

File: src/main/kotlin/com/lkimilhol/matchingproject/member/domain/Country.kt
package com.lkimilhol.matchingproject.member.domain

import com.lkimilhol.matchingproject.exception.NotFoundCountryException

enum class Country(val country: String) {
    KR("KR"),
    JA("JA");

    @Override
    fun valueOf():Country {
        when (name) {
            "KR" -> return KR
            "JA" -> return JA
        }
        throw NotFoundCountryException()
    }
}